python-checks/src/main/java/org/sonar/python/checks/RobustCipherAlgorithmCheck.java

```diff
@@ -63,14 +86,54 @@ public class RobustCipherAlgorithmCheck extends PythonSubscriptionCheck {
   public void initialize(Context context) {
     context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, subscriptionContext -> {
       CallExpression callExpr = (CallExpression) subscriptionContext.syntaxNode();
-      Symbol calleeSymbol = callExpr.calleeSymbol();
-      if (calleeSymbol != null) {
-        String fqn = calleeSymbol.fullyQualifiedName();
-        if (fqn != null && sensitiveCalleeFqns.contains(fqn)) {
-          subscriptionContext.addIssue(callExpr.callee(), MESSAGE);
-        }
-      }
+      Optional.ofNullable(callExpr)
+        .map(CallExpression::calleeSymbol)
+        .map(Symbol::fullyQualifiedName)
+        .filter(fqn -> sensitiveCalleeFqns.contains(fqn) || SSL_SET_CIPHERS_FQN.equals(fqn))
+        .ifPresent(str -> addMethodSpecificIssue(subscriptionContext, callExpr, str));
```

Comment:
I think here it would be easier to understand the program flow with branching: There are two case either it is a sensitive call and so we raise the issue or it is a ssl context call with a sensitive algorithm. I think it could be clearer to have your branching directly in the ifPresent.

Reply:
Thanks for the tip. I refactored this in 1dd32b3d33381b16d208e2d0d833a5b7c2956532.

Reply:
Thanks for the tip. I refactored this in 1dd32b3d33381b16d208e2d0d833a5b7c2956532.